driver-core/src/main/com/mongodb/internal/operation/AsyncQueryBatchCursor.java

```diff
@@ -190,12 +193,10 @@ private void next(final SingleResultCallback<List<T>> callback, final boolean tr
         if (isClosed()) {
             callback.onResult(null, new MongoException(format("%s called after the cursor was closed.",
                     tryNext ? "tryNext()" : "next()")));
-        } else if (firstBatch != null && (tryNext || !firstBatch.getResults().isEmpty())) {
-            // May be empty for a tailable cursor
-            List<T> results = firstBatch.getResults();
-            if (tryNext && results.isEmpty()) {
-                results = null;
-            }
+        } else if (firstBatch != null && (tryNext || !firstBatchEmpty)) {
```

Comment:
Is this part of the fix to the bug, or just a refactoring?   If just a refactoring, I suggest reverting it so as not to mix non-obvious refactoring with a bug fix.

Reply:
Reverted 0893c181da2e774cc8236760869b6b3ac140f775 in 94fc5e71e3e42ad1ea984399c852cdd5693d7bb1.